OpenRA.Mods.Cnc/Traits/Render/WithCargo.cs

```diff
@@ -45,6 +45,19 @@ public WithCargo(Actor self, WithCargoInfo info)
 			body = self.Trait<IBodyOrientation>();
 		}
 
+		public void Tick(Actor self)
+		{
+			if (facing == null)
+				return;
+
+			foreach (var c in cargo.Passengers)
+			{
+				var cargoFacing = c.TraitOrDefault<IFacing>();
```

Comment:
You can cache this when actors enter/leave the transport (compare with `AttackGarrisoned`) and avoid N trait lookups per tick.